plugins/x86/x86_operands.ml

```diff
@@ -14,10 +14,10 @@ let r ~f mem insn =
   | [| Op.Reg reg |] -> f mem reg
   | _ -> invalid_operands ~here:[%here] insn
 
-
 let i ~f mem insn =
   match Insn.ops insn with
   | [| Op.Imm imm |] -> f mem imm
+  | [| Op.Imm imm; Op.Reg reg |] when Reg.name reg = "Nil" -> f mem imm
```

Comment:
I think that this change is to global, why do you allow all instructions to be followed by a bogus Nil register? Is it really true that in x86 all instructions have this property? Probably, you should solve this issue local only for offending instructions.